steam_community_market/exceptions.py

```diff
@@ -1,25 +1,74 @@
-from steam_community_market.enums import ESteamUnsupportedCurrency
+from steam_community_market.currencies import SteamLegacyCurrency
+from typing import Any
 
 
-class SteamUnsupportedCurrency(Exception):
+class InvalidCurrencyException(Exception):
+    """Exception raised when a currency is considered to be invalid by the Steam Community Market.
+
+    This exception class is used to handle cases where a given currency is not
+    supported by the Steam Community Market.
+
+    .. versionadded:: 1.3.0
+
+    Attributes:
+        currency (Any): The unsupported invalid currency.
+
+    :param currency: The unsupported invalid currency.
+    :type currency: Any
+    """
+
+    def __init__(self, currency: Any) -> None:
+        """Initialize the InvalidCurrencyException with the unsupported invalid currency.
+
+        :param currency: The unsupported invalid currency.
+        :type currency: Any
+        """
+
+        self.currency = currency
+
+    def __str__(self) -> str:
+        """Return a string representation of the exception.
+
+        This method returns a string that indicates which currency is not supported
+        by the Steam Community Market.
+
+        :return: A string representation of the exception.
+        :rtype: str
+        """
+
+        return f'Currency "{self.currency}" is not supported by the Steam Community Market.'
```

Comment:
Typically you would just use `super().__init__(f'Currency "{self.currency}" is not supported by the Steam Community Market.')` and I wouldn't personally bother this this method and docstring

Reply:
From my perspective it would be nice to have everything documented. @offish, what do you say?